bundles/org.openhab.binding.qbus/src/main/java/org/openhab/binding/qbus/internal/handler/QbusBistabielHandler.java

```diff
@@ -0,0 +1,210 @@
+                    "Error while initializing the thing.");
+        }
+    }
+
+    /**
+     * Returns the serial number of the controller
+     *
+     * @return the serial nr
+     */
+    public @Nullable String getSN() {
+        return this.sn;
+    }
+
+    /**
+     * Sets the serial number of the controller
+     */
+    public void setSN() {
+        QbusBridgeHandler qBridgeHandler = getBridgeHandler("Bistabiel", bistabielId);
+        if (qBridgeHandler == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
+                    "No communication with Qbus Bridge!");
+            return;
+        }
+        this.sn = qBridgeHandler.getSn();
+    }
+
+    /**
+     * Handle the status update from the thing
+     */
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        QbusCommunication qComm = getCommunication("Bistabiel", bistabielId);
+        if (qComm == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Bridge communication not initialized when trying to execute command for bistabiel " + bistabielId);
+            return;
+        }
+
+        Map<Integer, QbusBistabiel> bistabielComm = qComm.getBistabiel();
+
+        if (bistabielComm != null) {
+            QbusBistabiel qBistabiel = bistabielComm.get(bistabielId);
+
+            if (qBistabiel == null) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "bridge communication not initialized when trying to execute command for bistabiel "
+                                + bistabielId);
+                return;
+            } else {
+                scheduler.submit(() -> {
+                    if (!qComm.communicationActive()) {
+                        restartCommunication(qComm, "Bistabiel", bistabielId);
+                    }
+
+                    if (qComm.communicationActive()) {
+                        if (command == REFRESH) {
+                            handleStateUpdate(qBistabiel);
+                            return;
+                        }
+
+                        handleSwitchCommand(qBistabiel, channelUID, command);
+                    }
+                });
+            }
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
+                    "Error while initializing the thing.");
+        }
+    }
+
+    /**
+     * Executes the switch command
+     *
+     * @throws InterruptedException
+     */
+    private void handleSwitchCommand(QbusBistabiel qBistabiel, ChannelUID channelUID, Command command) {
+        if (command instanceof OnOffType) {
+            OnOffType s = (OnOffType) command;
+            String snr = getSN();
+            if (snr != null) {
+                if (s == OnOffType.OFF) {
```

Comment:
The command don't need to be cast.
```suggestion
            String snr = getSN();
            if (snr != null) {
                if (command == OnOffType.OFF) {
```